js/src/redux/providers/chainMiddleware.js

```diff
@@ -24,20 +22,34 @@ export default class ChainMiddleware {
       if (action.type === 'statusCollection') {
         const { collection } = action;
 
-        if (collection && collection.netChain) {
-          const newChain = collection.netChain;
+        if (collection) {
           const { nodeStatus } = store.getState();
+          const { netChain, nodeKind } = nodeStatus;
+          const newChain = collection.netChain;
+          const newNodeKind = collection.nodeKind;
+          let reloadChain = false;
+          let reloadType = false;
+
+          // force reload when chain has changed and is not initial value
+          if (newChain) {
+            const hasChainChanged = newChain !== netChain;
+            const isInitialChain = netChain === DEFAULT_NETCHAIN;
+
+            reloadChain = !isInitialChain && hasChainChanged;
```

Comment:
Maybe it is slightly wrongly named. On startup we don't know the chain, so we set it to `DEFAULT_CHAIN` (`DEFAULT_CHAIN = '(unknown)'`). So when the value changes from this, we don't do the reload, but on any subsequent changes, we do.